### PR TITLE
TG-97: Endpoint response normalization and test

### PR DESCRIPTION
**Pull request by martinalbrnz**
Create -> 201
Delete -> 204
Get -> 200
Not found -> 404
Bad request -> 400

successfull response:
{
  message: 'X has been found/updated/deleted',
  data: {data del sv},
  error: false,
}

bad response:
{
  message: error.message, (generated by JOI)
  data: {},
  error: true,
}

validation error response {
  message: 'There has been an error in the validation',
  data: validation error msg (E.G.: '\"active\" is required'),
  error: true,
}

Error is a boolean

Messages:
  getAll: 'All X are:'
  getById: 'X with ID:${id} sent:'
  create: 'X has been created'
  update: 'X successfully updated'
  delete: 'X successfully deleted'
  notFound: 'X with ID:${id} not found'
  missingId: 'Missing ID parameter'

X is referred to Admin, Superadmin, etc...

Jest tests has been fixed according to new requirements

### CODE REVIEW

File: src/controllers/projects.js [hunk 1]
```diff
@@ -20,24 +22,26 @@ const getAllProjects = async (req, res) => {
 const getProjectById = async (req, res) => {
   try {
     const projectId = req.params.id;
-    const project = await Project.findOne({ _id: projectId }).populate('team', { firstName: 1, surname: 1 }).populate('tasks', { taskName: 1 });
+    const project = await Project.findOne({ _id: projectId })
+      .populate('team', { firstName: 1, lastName: 1 })
```
Comment (mpueblas) [inline src/controllers/projects.js:26]: it`s a doubt, not a correction, but where 'team' comes from to be populated?

**Review by mpueblas — APPROVED** [review]

File: src/controllers/projects.js (hunk 1 above)
Comment (martinalbrnz) [reply src/controllers/projects.js:26]: 'team' is the name of the variable holding an array of employee's id.
If you look at the Projects model, there is a 'team' field which have a ref: 'Employee', indicating that this field should be populated with items from Employee
```
team: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Employee',
      },
    ],
```
When it is populated, you end up with an array of objects like this: { _id, firstName, lastName }
something like this:
 ```
[
  { _id, firstName, lastName }, 
  { _id, firstName, lastName }, 
  { _id, firstName, lastName }
]
```

**Review by AlfonsoDalix — APPROVED** [review]
Good job! Looks quite nice

**Review by karu-ams — APPROVED** [review]
## Testing Results
_Tested locally on POSTMAN with an enviroment's variable with URL: https://alfonso-trackgenix-server-ho6vc6jh9-basp-m2022.vercel.app/_

Test Results: PASS ✅ READY TO MERGE

**Pull request merged**